lib/bq_stream/bq_stream.rb

```diff
@@ -98,21 +101,47 @@ def dequeue_items
       end
       create_bq_writer
       # Batch sending to BigQuery is limited to 10_000 rows
-      records =
+      # Initiate the records to be compared to the data that will be sent to BigQuery
+      prep_records =
         QueuedItem.where(sent_to_bq: nil).where('updated_at < ?', dequeue_time).limit([batch_size, 10_000].min)
-      data = records.collect do |i|
+      # Create data hash that will be sent to BigQuery  
+      data = prep_records.collect do |i|
         new_val = encode_value(i.new_value) rescue nil
         { table_name: i.table_name, record_id: i.record_id, attr: i.attr,
           new_value: new_val ? new_val : i.new_value, updated_at: i.updated_at }
       end
+      records =[]
+      # Compare prep_records to data and create a new array of the records actually being sent to BigQuery
+      unless prep_records.empty?
```

Comment:
If an array is empty each just won't do anything, so should be no need for extra conditional